File: objc/Project5/Project5/Pin.m
//
//  Pin.m
//  Project5
//
//  Created by Jinwoo Kim on 12/24/20.
//

#import "Pin.h"

@implementation Pin
+ (Pin *)title:(NSString *)title coordinate:(CLLocationCoordinate2D)coordinate color:(NSColor *)color {
    Pin *new = [Pin new];
    new.title = title;
    new.coordinate = coordinate;
    new.color = color;
    return new;
}

+ (Pin *)title:(NSString *)title coordinate:(CLLocationCoordinate2D)coordinate {
    Pin *new = [Pin
                title:title
                coordinate:coordinate
                color:NSColor.greenColor];
    return new;
}
@end

File: swift/Project5/Project5/Pin.swift
//
//  Pin.swift
//  Project5
//
//  Created by Jinwoo Kim on 12/23/20.
//

import Cocoa
import MapKit

class Pin: NSObject, MKAnnotation {
    var title: String?
    var subtitle: String?
    var coordinate: CLLocationCoordinate2D
    var color: NSColor
    
    init(title: String, coordinate: CLLocationCoordinate2D, color: NSColor = .green) {
        self.title = title
        self.coordinate = coordinate
        self.color = color
    }
}
